fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_life.dm

```diff
@@ -330,27 +346,29 @@
 			if(owner.current.AmStaked())
 				to_chat(owner.current, "<span class='userdanger'>You are staked! Remove the offending weapon from your heart before sleeping.</span>")
 				return
-			/// Otherwise, check for Sol, or if injured enough to enter Torpor.
-			if(clan.bloodsucker_sunlight.amDay || total_damage >= 10)
-				to_chat(owner.current, "<span class='notice'>You enter the horrible slumber of deathless Torpor. You will heal until you are renewed.</span>")
-				Torpor_Begin()
-	/// If it's not Sol and you have 0 Brute damage, check to End Torpor.
-	if(!clan.bloodsucker_sunlight.amDay && total_brute <= 10)
-		if(HAS_TRAIT(owner.current, TRAIT_NODEATH))
+			/// Otherwise, check if it's Sol, to enter Torpor.
+			if(clan.bloodsucker_sunlight.amDay && !HAS_TRAIT(owner.current, TRAIT_NODEATH))
+				Check_Begin_Torpor(TRUE)
+		/// You are in Torpor, and in a Coffin. Check if it's not Daytime & you have less than 10 Brute/Burn combined to end Torpor.
+		else if(!clan.bloodsucker_sunlight.amDay && total_damage <= 10)
 			Check_End_Torpor()
-
-/datum/antagonist/bloodsucker/proc/Torpor_Begin()
-	/// Force them to go to sleep
-	REMOVE_TRAIT(owner.current, TRAIT_SLEEPIMMUNE, BLOODSUCKER_TRAIT)
-	/// Without this, you'll just keep dying while you recover.
-	ADD_TRAIT(owner.current, TRAIT_NODEATH, BLOODSUCKER_TRAIT)
-	ADD_TRAIT(owner.current, TRAIT_FAKEDEATH, BLOODSUCKER_TRAIT)
-	ADD_TRAIT(owner.current, TRAIT_DEATHCOMA, BLOODSUCKER_TRAIT)
-	owner.current.Jitter(0)
-	/// Disable ALL Powers
-	for(var/datum/action/bloodsucker/power in powers)
-		if(power.active && !power.can_use_in_torpor)
-			power.DeactivatePower()
+			return
+	/// You're not in a Coffin, but are in Torpor. Check if it's not Daytime, & you have less than 10 Brute (NOT Burn) to end Torpor.
+	if(!clan.bloodsucker_sunlight.amDay && total_brute <= 10 && HAS_TRAIT(owner.current, TRAIT_NODEATH))
+		Check_End_Torpor()
+
+/datum/antagonist/bloodsucker/proc/Check_Begin_Torpor(CheckChecks = FALSE)
+	/// Are we entering Torpor via Sol/Death? Then entering it isnt optional!
+	if(CheckChecks)
+		Torpor_Begin()
+		return
+	var/mob/living/carbon/user = owner.current
+	var/total_brute = user.getBruteLoss_nonProsthetic()
+	var/total_burn = user.getFireLoss_nonProsthetic()
+	var/total_damage = total_brute + total_burn
+	/// Checks - Not daylight & Has more than 10 Brute/Burn & not already in Torpor
+	if(!clan.bloodsucker_sunlight.amDay && total_damage >= 10 && !HAS_TRAIT(owner.current, TRAIT_NODEATH))
+		Torpor_Begin()
 
 /datum/antagonist/bloodsucker/proc/Check_End_Torpor()
 	/// You're not in Sol? (Slept in a Locker for example), then you don't need to leave it.
```

Comment:
"You're not in Sol?" -> "You're not in torpor?"

Reply:
No, you're not in Sol. You can sleep through Sol in a locker.

Reply:
Comment was made before the changes, oopsies

Reply:
Or wait no, this was referring to line 374, which is collasped by default. The comment "You're not in sol" on 373 is above an if statement that checks for NODEATH on line 374